src/Campaigns/CampaignSubscriptionQuery.php

```diff
@@ -0,0 +1,109 @@
+<?php
+
+namespace Give\Campaigns;
+
+use DateTimeInterface;
+use Give\Campaigns\Models\Campaign;
+use Give\Donations\ValueObjects\DonationMetaKeys;
+use Give\Framework\QueryBuilder\JoinQueryBuilder;
+use Give\Framework\QueryBuilder\QueryBuilder;
+
+/**
+ * @unreleased
+ */
+class CampaignSubscriptionQuery extends QueryBuilder
+{
+    /**
+     * @unreleased
+     */
+    public function __construct(Campaign $campaign)
+    {
+        $this->from('give_subscriptions', 'subscription');
+
+        // Include only current payment "mode"
+        $this->where('payment_mode', give_is_test_mode() ? 'test' : 'live');
+
+        // Include only valid statuses
+        $this->joinDonation('donation');
+        $this->whereIn('donation.post_status', ['publish', 'give_subscription']);
+
+        // Include only forms associated with the Campaign.
+        $this->joinDonationMeta(DonationMetaKeys::CAMPAIGN_ID, 'campaignId');
+        $this->where('campaignId.meta_value', $campaign->id);
+
+    }
+
+    /**
+     * @unreleased
+     */
+    public function between(DateTimeInterface $startDate, DateTimeInterface $endDate): self
+    {
+        $query = clone $this;
```

Comment:
The only thing i'm not sure about is why we are cloning the query for a lot of these methods.  I saw this pattern in the `CampaignDonationsQuery` and it confused me.  It doesn't have to hold up the PR but something to think about if it's necessary.

Reply:
Yeah, honestly, I don’t think it’s necessary. Though, it’s been added to the `CampaignDonationQuery` methods for some reason, so I kept that pattern. Would you like me to remove those?

Reply:
yeah I think it's safe to remove them.  I don't like cloning objects unless its absolutely necessary and I'm not sure why it would be needed here.

Reply:
All removed on https://github.com/impress-org/givewp/pull/7786/commits/d99d34bf2d3c9e2e398dfe45ad5b26a74393854d

Reply:
@jonwaldstein Oh, I see what’s happening here. Those methods are not just meant to append instructions to the main query but to execute it and return a result. By cloning the query, we can use the same CampaignDonationQuery object to perform multiple requests with methods like `countDonations`, `countDonors`, and so on. Otherwise, we would need to create a new object for each request. I think we'll need to revert that commit back.

Reply:
ah I see